async-inn/Controllers/AmenitiesController.cs

```diff
@@ -79,6 +84,7 @@ public async Task<IActionResult> PutAmenity(int id, Amenity amenity)
 
         // POST: api/Amenities
         // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        [Authorize(Roles = "District Manager, Property Manager, Agent")]

```

Comment:
The way I interpret the specs is that a Property Manager and Agent can add an amenity to a room (as in the route AddAmenityToRoom in your Rooms Controller) But that ultimately only a District Manager can create a new amenity.